src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/blocks/HologramProjector.java

```diff
@@ -119,14 +147,59 @@ private void openEditor(@Nonnull Player p, @Nonnull Block projector) {
             return false;
         });
 
-        menu.addItem(1, new CustomItemStack(Material.CLOCK, "&7Offset: &e" + NumberUtils.roundDecimalNumber(Double.valueOf(BlockStorage.getLocationInfo(projector.getLocation(), OFFSET_PARAMETER)) + 1.0D), "", "&fLeft Click: &7+0.1", "&fRight Click: &7-0.1"));
+        menu.addItem(1, new CustomItemStack(Material.CLOCK, "&7Offset X: &e" + getXOffset(projector), "", "&fLeft Click: &7+0.1", "&fRight Click: &7-0.1"));
+        menu.addMenuClickHandler(1, (pl, slot, item, action) -> {
+            double offsetX = NumberUtils.reparseDouble(getXOffset(projector) + (action.isRightClicked() ? -0.1F : 0.1F));
+            double offsetY = getYOffset(projector);
+            double offsetZ = getZOffset(projector);
+
+            if (offsetX < -MAX_HORIZONTAL_OFFSET) {
+                offsetX = -MAX_HORIZONTAL_OFFSET;
+            } else if (offsetX > MAX_HORIZONTAL_OFFSET) {
+                offsetX = MAX_HORIZONTAL_OFFSET;
+            }
+            
+            ArmorStand hologram = getArmorStand(projector, true);
+            Location l = new Location(projector.getWorld(), projector.getX() + offsetX, projector.getY() + offsetY, projector.getZ() + offsetZ);
+            hologram.teleport(l);
+
+            BlockStorage.addBlockInfo(projector, OFFSET_PARAMETER_X, String.valueOf(offsetX));
+            openEditor(pl, projector);
+            return false;
+        });
+
+        menu.addItem(1, new CustomItemStack(Material.CLOCK, "&7Offset Y: &e" + getYOffset(projector) + 1.0D, "", "&fLeft Click: &7+0.1", "&fRight Click: &7-0.1"));
         menu.addMenuClickHandler(1, (pl, slot, item, action) -> {
-            double offset = NumberUtils.reparseDouble(Double.valueOf(BlockStorage.getLocationInfo(projector.getLocation(), OFFSET_PARAMETER)) + (action.isRightClicked() ? -0.1F : 0.1F));
+            double offsetX = getXOffset(projector);
+            double offsetY = NumberUtils.reparseDouble(getYOffset(projector) + (action.isRightClicked() ? -0.1F : 0.1F));
```

Comment:
Is there any limit to the yOffset like how you have the x and z limited? If not I think one should be added for both consistency but also just it makes sense

Reply:
Currently in SF there's no limit, and realistically anything over 10 blocks away is annoyingly hard to see to line up anyway. I wouldn't want to add a limit on something currently unlimited either just incase someone clicks the button to move it up 0.1 and suddenly it's 10 blocks higher and can't go back down